### DISCUSSION TITLE
[Android] [Auth] Random error 10

### DISCUSSION THREAD

**Discussion by kapik** [capawesome-team/capacitor-firebase#415] [Q&A]
Hello everyone,

I've been having problems for a few days now.
I don't understand where my error could be coming from: I get error 10 randomly when I try to connect with Google.

Here's an example:
- I build my Angular/Ionic application
- I build the Android application (via Android Studio)
- I test: it works 
- I build my Angular/Ionic application again without any changes
- I build the Android application (via Android Studio)
- I test: it doesn't work anymore
- ... 

I've reduced my application as much as possible: only 1 component that calls the method: 
```ts

public async google(): Promise<void> {
    try {
      const result = await FirebaseAuthentication.signInWithGoogle()
      alert(result?.user?.uid)
    } catch (error) {
      alert('💥')
    }
  }
```

So I guess I don't have a problem with SHA1.
Do you have any ideas?

I'm using AngularFire which I configure in my `appModule`:
```ts 

@NgModule({
  declarations: [
    AppComponent
    LoginComponent
  ],
  imports: [
    AngularPipesModule,
    CommonModule,
    FormsModule,
    BrowserModule,
    BrowserAnimationsModule,
    HttpClientModule,
    RouterModule.forRoot([{
      path: 'home',
      component: LoginComponent
    },{
      path: '',
      redirectTo: 'home',
      pathMatch: 'full'
    }]),
    provideFirebaseApp(() => initializeApp(environment.firebase)),
    provideFirestore(() => getFirestore()),
    provideAuth(() => getAuth(getApp())),

    IonicModule.forRoot()
  ],
  providers: [
    { provide: RouteReuseStrategy, useClass: IonicRouteStrategy }
  ],
  bootstrap: [AppComponent]
})
```

Here is my capacitor config: 

```ts
...
FirebaseAuthentication: {
      skipNativeAuth: false,
      providers: ['google.com']
    }
...
```
I'm pre-empting the request for code to reproduce the bug: I'm going to see about redoing a project from scratch.
I hope you can help me, I can't look any further 🙏

**Comment by robingenz** [comment]
Hi @kapik,
your issue should be a duplicate of https://github.com/robingenz/capacitor-firebase-authentication/issues/97.

**Reply by kapik** [reply]
Hey @robingenz 
I've already read these issues :( 
but it's not a SHA1 problem as it works sometimes

**Reply by robingenz** [reply]
So when does it work and when doesn't it work?

**Reply by kapik** [reply]
I don't know. That's precisely my problem.
A rebuild of the app without changing the source code or the Firebase config has different results

**Reply by kapik** [reply]
I work in a monorepo. I'm trying to extract my app to have something to show 😬

**Reply by kapik** [reply]
I'm gradually lightening my repo 😅
I've just redone a test by generating the apk from Android Studio.
I checked the SHA1 with the command: `keytool -printcert -jarfile apps/story-android/app/release/app-release.apk` and this SHA1 is in my firebase. But I still get the error.
I'm thinking that there must be another config somewhere, either missing or wrong.

**Comment by Jelberg** [comment]
I have the same problem with Google authentication
I added both the SHA1 and SHA256 I got by console to Firebase but I have the same error.

I use capacitor and Ionic, add the variable that says in the documentation in strings.xml and the google-services.json file update it and also add the FirebaseAuthentication parameters in capacitor.config.ts

I'm not sure it's a configuration problem because I use the login with user and password with the firebase/auth library and I have no problems with it

**Reply by robingenz** [reply]
Hey, it's definitely a configuration error. I use the latest plugin version in my production apps. Password authentication does not require sha1/sha256.

**Reply by Jelberg** [reply]
Hi, thanks for replying. I saw what the error. 

Android Studio sign a debug certificate automatically. The key is stored in ~/.android/debug.keystore and that's where I get the fingerprint to add in Firebase.

With that it worked properly!

Now this generated another question, why does not take into account the .jks that I generate? Or only takes them into account in production?

**Reply by robingenz** [reply]
You can configure that in your build.gradle file, see https://stackoverflow.com/a/19512364/6731412
